nipype/algorithms/modelgen.py

```diff
@@ -144,6 +144,61 @@ def scale_timings(timelist, input_units, output_units, time_repetition):
     timelist = [np.max([0., _scalefactor * t]) for t in timelist]
     return timelist
 
+def bids_gen_info(bids_event_files,
+    condition_column='trial_type',
+    amplitude_column=None,
+    time_repetition=False,
+    ):
+    """Generate subject_info structure from a list of BIDS .tsv event files.
+
+    Parameters
+    ----------
+
+    bids_event_files : list of str
+        Filenames of BIDS .tsv event files containing columns including:
+        'onset', 'duration', and 'trial_type' or the `condition_column` value.
+    condition_column : str
+        Column of files in `bids_event_files` based on the values of which
+        events will be sorted into different regressors
+    amplitude_column : str
+        Column of files in `bids_event_files` based on the values of which
+        to apply amplitudes to events. If unspecified, all events will be
+        represented with an amplitude of 1.
+
+    Returns
+    -------
+
+    list of Bunch
+    """
+    info = []
+    for bids_event_file in bids_event_files:
+        with open(bids_event_file) as f:
+            f_events = csv.DictReader(f, skipinitialspace=True, delimiter='\t')
+            events = [{k: v for k, v in row.items()} for row in f_events]
+            conditions = list(set([i[condition_column] for i in events]))
```

Comment:
You can release the file handle as soon as it has been fully read.

```suggestion
        conditions = list(set([i[condition_column] for i in events]))
```